jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunExecutorActor.scala

```diff
@@ -47,35 +48,31 @@ class JobRunExecutorActor(
 
   override def preStart(): Unit = {
     val startingDeadline = computeStartingDeadline
-    if (startingDeadline.exists(d => d.toSeconds <= 0)) {
-      log.info(s"StartingDeadline timeout of ${jobRun.startingDeadline.get} expired for JobRun ${jobRun.id} created at ${jobRun.createdAt}")
-      becomeAborting()
-    } else {
-      startingDeadline.foreach(scheduleStartingDeadline)
-
-      jobRun.status match {
-        case JobRunStatus.Initial => becomeCreating()
-
-        case JobRunStatus.Starting | JobRunStatus.Active =>
-          launchQueue.get(runSpecId) match {
-            case Some(info) if info.finalTaskCount > 0 =>
-              log.info("found an active launch queue for {}, not scheduling more tasks", runSpecId)
-              val tasks = taskTracker.appTasksLaunchedSync(runSpecId).collect {
-                // FIXME: we do currently only allow non-resident tasks. Since there is no clear state on
-                // Marathon's task representation, this is the safest conversion we can do for now:
-                case task: LaunchedEphemeral => JobRunTask(task)
-                case task: Task              => throw UnexpectedTaskState(task)
-              }
-              self ! Initialized(tasks)
-
-            case _ =>
-              self ! Initialized(tasks = Nil)
-          }
-
-        case JobRunStatus.Success => becomeFinishing(jobRun)
-
-        case JobRunStatus.Failed  => becomeFailing(jobRun)
-      }
+    startingDeadline.foreach(scheduleStartingDeadline)
+
+    jobRun.status match {
+      case JobRunStatus.Initial | JobRunStatus.Starting if startingDeadline.exists(d => d.toSeconds <= 0) =>
+        log.info(s"StartingDeadline timeout of ${jobRun.startingDeadline.get} expired for JobRun ${jobRun.id} created at ${jobRun.createdAt}")
+        becomeAborting()
+      case JobRunStatus.Initial =>
+        becomeCreating()
+      case JobRunStatus.Starting | JobRunStatus.Active =>
+        val existingTasks = tasksFromTaskTracker()
+        existingTasks match {
+          case tasks if tasks.nonEmpty =>
+            // the actor was probably just restarted and we did not lose contents of the launch queue
+            self ! Initialized(tasks)
+          case _ if jobRun.status == JobRunStatus.Starting && existsInLaunchQueue() =>
```

Comment:
is my assumption correct, that the situation for item being in launch queue but not in task tracker can happen only for starting task?

Reply:
Yes